### DISCUSSION TITLE
Merge queue should allow to decide whether to squash or not in the PR itself

### DISCUSSION THREAD

**Discussion by nazar-pc** [community/community#68723] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

I'd like for merge queue, just like auto-merge, support decision about how to merge in the PR itself.

We generally prefer merge, but sometimes we end up with PR that in the end has a bunch of commits, but changes very little, in which case we do squash-merge. With merge queue right now this would require manual squashing and force-pushing.

**Comment by notxcain** [comment]
Yes, please!

At our company we use a monorepo approach and different teams have different branching styles. 
Some prefer having a branch that accumulates squashed PRs for a certain release. To preserve these squashed PR commits when merging into `main`, they use a simple merge strategy.
Other teams squash their PR directly to `main` as soon as they're ready.

We really would like to use the merge queue feature, but having a single merge strategy for all PRs makes it impossible.

**Comment by aa333** [comment]
Another popular case in our company:
We enforce squash when merging into develop branch to get a nice history of 1 PR = 1 commit. 

However, when back-merging hotfixes from the release branches, we already have granular commits history which right now gets squashed again. And there seems to be no bypass for that.

**Comment by tomage** [comment]
I would like this as well.

Most people in my company do "Squash and merge", whilst I prefer to tidy up my commit history myself using interactive rebasing. Sometimes, I have a PR that has more than one logical change (I know, I _should_ probably just do it in multiple PRs - but one can argue sometimes it's worth doing multi-commit PR), so it's annoying to not have the option to choose when using merge queues.

**Comment by branberry** [comment]
I'd love to use merge queue functionality, but the fact that you can't squash and merge is just a complete deal breaker. Is there a reason why this isn't just the default? Most repos use squash and merge for PRs.

Many branches that I work on can have dozens of commits, and this totally pollutes the main branch, making it much more difficult for new developers to understand changes we make when looking back at the commit history. Does anyone know if this is functionality that's on the roadmap? I would love to use this feature in some of my repos.

**Reply by nazar-pc** [reply]
You misunderstood what this request is about. You totally can use merge queue with squash-merge. What you can't do is decide whether to squash or not on each individual merge.

**Reply by branberry** [reply]
Oh, got it! I totally misunderstood this, I got egg on my face from that one. Thank you for the clarification!

**Comment by samlaf** [comment]
+1 for this. makes merge queue unusable for our org.
